### ISSUE TITLE
Connection timeout issues since upgrade from 8.0 to 8.4

### ISSUE THREAD

**Issue by MatanYadaev** [laravel/scout#427]
- Scout Version: 8.4.0
- Laravel Version: 8.13.0
- PHP Version: 7.4.11
- Database Driver & Version: MySQL 5.7, Redis 6

### Description:
Since I've upgraded from Laravel 7 to 8 and from Scout 8.0 to 8.4 I began to have timeout issues when indexing entries in the queue (using `MakeSearchable`).
It happens both on local and on staging. (didn't deploy to production because of this issue)

The issues as captured by Sentry:
```
GuzzleHttp\Exception\ConnectException
cURL error 28: Resolving timed out after 2000 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://APP_ID.algolia.net/1/indexes/INDEX_ID/batch
```
```
GuzzleHttp\Exception\ConnectException
cURL error 28: SSL connection timeout (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://APP_ID.algolia.net/1/indexes/INDEX_ID/batch
```

A few notes:
1. It happens on some indexes, not only a specific one.
2. The index is not really a batch, it's one by one.
3. It's not consistent. It's not 100% failing all the time. Only sometimes.

### Steps To Reproduce:
I don't know how to reproduce it. It's not consistent, and I'm not sure if it will happen on a new Laravel project. My project is a few years old, much code, many packages. It's hard to point what responsible for the issue.

**Comment by driesvints** [comment]
Since this is the only report we got so far for this and the project is widely used I think it's better that you try a support channel first. If you can verify that this is a bug in Scout itself feel free to report back.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks

**Comment by Demers94** [comment]
@MatanYadaev Did you find a solution for this?

I recently upgraded an app from Laravel 5.8 to Laravel 8 and I started having this problem. It only happens when I'm running the app in Docker (using Laravel Sail)

**Comment by Demers94** [comment]
@MatanYadaev Did you manage to fix this?

I recently upgraded a project from Laravel 5.8 to Laravel 8 and I started having this issue. 

It only happens when I'm running the app in Docker (using Laravel Sail)

**Comment by MatanYadaev** [comment]
@Demers94 No, I still have it. Both for local environments (non-Docker) and production.

**Comment by chloelbn** [comment]
Hi @driesvints, we're having several report of customers experiencing this issue since the last few days (ie https://discourse.algolia.com/t/laravel-timeout-reimport/11651). Did you find anything on your side that would explain the timeout?

**Comment by driesvints** [comment]
We really need a way to reproduce it before we can check in on this.

**Comment by sal1ner0** [comment]
@driesvints it happens almost always when i run scout:reimport directly on terminal or through laravel scheduler.
The problem increases because most of the time it deletes the index and we run out of articles...

**Comment by driesvints** [comment]
when i run `scout:reimport`

There's no such command in Scout. You're probably talking about https://github.com/algolia/scout-extended/blob/master/src/Console/Commands/ReImportCommand.php ?

If that's the case please open up an issue on the scout extended repository. If you can't reproduce it with just Scout itself then we can't really help out here.

**Comment by kossa** [comment]
I have the same problem on the server but works fine on local with the same version

I guess problem with server config

**Comment by sal1ner0** [comment]
OK thanks @driesvints , i just opened a new issue there... 
[](https://github.com/algolia/scout-extended/issues/263)
Hope they can help me..

**Comment by mfullbrook** [comment]
[This is the issue](https://github.com/algolia/scout-extended/issues/263) created by @sal1ner0 over on algolia/scout-extended where there is a helpful discussion as well as Algolia's diagnostic script.

We had the same symptons today.   After contacting Algolia support they confirmed there was an issue with the search cluster our indexes were on.

**Comment by louisjeck** [comment]
Same issue here, using 
`"laravel/scout": "8.6.0"`, `"algolia/scout-extended": "1.13.0"`, `"laravel/framework": "8.27.0"`. I'm getting random timeout issues, even on searching, not only indexing. I've notice that once it timed out, it becomes responsive again for a few seconds to a minute. Using Laravel Sail aswell. I'm noticing the same issue with order external APIs, first search is always really slow / times out.

**Comment by jadamec** [comment]
I've same issue...

**Comment by jadamec** [comment]
> Same issue here, using
> `"laravel/scout": "8.6.0"`, `"algolia/scout-extended": "1.13.0"`, `"laravel/framework": "8.27.0"`. I'm getting random timeout issues, even on searching, not only indexing. I've notice that once it timed out, it becomes responsive again for a few seconds to a minute. Using Laravel Sail aswell. I'm noticing the same issue with order external APIs, first search is always really slow / times out.

Absolutely the same...
